src/charm.py

```diff
@@ -249,7 +254,7 @@ def _validate(self):
         if not self._state.schema_ready:
             raise ValueError("admin:temporal relation: schema is not ready")
 
-        if self.config["auth-enabled"]:
+        if self.config["auth-enabled"] and "frontend" in self.config["services"]:
```

Comment:
This is fine, but it's not immediately apparent to me why "frontend" needs to be in `self.config["services"]`. But before the change, checking if `auth-enabled` is true was very easy to understand.